src/rust_api.rs

```diff
@@ -243,9 +243,13 @@ impl TryFrom<&[u8]> for DnaSequence {
     type Error = TranslationError;
 
     fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
-        let mut vec = vec![Nucleotide::N; value.len()];
-        for (idx, &b) in value.iter().enumerate() {
-            vec[idx] = Nucleotide::try_from(b)?;
+        let mut vec = vec![];
+        vec.reserve(value.len());
+
+        for &b in value {
+            if b != b' ' {
```

Comment:
I think we should also allow at least `b'\t'` and maybe `b'\n'` as well?

Reply:
Yes, as I've said several times now in Zulip *and* in the spec on the wiki, we should allow (ASCII) space \n\r\t, e.g., all the obvious ASCII whitespace (four separate characters).  If both of \n and \r are already being stripped out upstream, we can check for just ' ' and '\t' here.  We should, however, probably *not* use some [[:space:]] sort of regexp, because I don't have any idea what that might do in a Unicode locale, and we shouldn't be allowing Unicode in sequences but only in '>' lines.

(At the risk of allowing \v and \f and generally attempting to over-optimize when the compiler may not care, we *could* allow any ASCII code from #x08 through #0D inclusive, plus #20, but that's probably overkill.)